lib/scnnr/connection.rb

```diff
@@ -5,11 +5,16 @@ class Connection
     require 'net/http'
     require 'json'
 
-    def initialize(uri, method, api_key, logger)
+    MAX_SLEEP_ON_RETRY = 5
+    MAX_RETRY_LIMIT = 15
+
+    def initialize(uri, method, api_key, options)
```

Comment:
I encountered a similar situation about options hashes in another PR. Though I'm not sure if I can link the comments here directly, because this project is open source.
To avoid linking, I'll summarize it here:
- There is an interesting article first class keywords: https://thoughtbot.com/blog/ruby-2-keyword-arguments#keyword-arguments-vs-options-hash -> Would it be better to avoid option hashes?
- And I guess the reason for choosing the options hash in the first place was, that there are many arguments (I did the exact same thing before, lol). Does that mean, this method / object has too many responsibilities?

Reply:
@iusztin 
> And I guess the reason for choosing the options hash in the first place was, that there are many arguments

Exactly :smile: However I don't feel like I've extended responsibility of this method/object. Previously it was receiving `logger` which is one of the options that can be configured. I've just added another options that can be used in class. So there was no introduction of new responsibility. I had to use hash here to keep number of arguments below 5 :robot: 

Using keywords - I agree, probably it would be better to do so. However in this case I couldn't think of other way out. Am I missing something here?

Reply:
> I've just added another options that can be used in class. So there was no introduction of new responsibility. 

Isn't "assign a parameter to a state" a responsibility?

> I had to use hash here to keep number of arguments below 5

What do you think is the reason for that Rubocop rule? And is adding a hash to allow more arguments following the spirit of that rule?

>  However in this case I couldn't think of other way out. Am I missing something here?

In my opinion you already solved it after the latest push, by using constants :) So this point is OK for me 👍